### DISCUSSION TITLE
How to check the VCPKG_LIBRARY_LINKAGE of the port in project cmake file?

### DISCUSSION THREAD

**Discussion by vipcxj** [microsoft/vcpkg#37193] [Q&A] [answered]
How to check the VCPKG_LIBRARY_LINKAGE of the port in project cmake file?

Many ports provide different packages depending on whether they are dynamically or statically linked, which results in triplet-related project cmake files. This obviously doesn't make sense. So I want to be able to determine the linkage of the port in the project cmake file to import different packages. how can I do that?

**Comment by Osyotr** [comment]
Generally not possible. As a consumer, you search for packages, not static or dynamic libraries. What's your specific use case?

**Reply by vipcxj** [reply]
I only depend on a couple of libraries right now and 2 of them already have this problem. One is libdatachannel, which is dynamically linked by default for windows, so on my windows computer my cmake file has to say this: 
```
target_link_libraries(main PRIVATE LibDataChannel::LibDataChannel)
```
while the linux defaults to static linking, so when my project is compiled on linux it must be written like this:
```
target_link_libraries(test1 PRIVATE LibDataChannel::LibDataChannelStatic)
```

The other one is gstreamer. for static linking I can import gstreamer-full-1.0, while for dynamic linking there is no such import item. Am I the only one experiencing this problem?

**Reply by Neumann-A** [reply, marked as answer]
You can check if a target exists and link it accordingly: 
`$<TARGET_NAME_IF_EXISTS:tgt>` or and `if(TARGET <targetname>)`

**Comment by dg0yt** [comment]
Cf. zstd:

~~~cmake
    target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:zstd::libzstd_shared>,zstd::libzstd_shared,zstd::libzstd_static>)
~~~
